### PR TITLE
[Enhance] Support training with combined datasets

### PR DESCRIPTION
**Pull request by Ben-Louis**
## Motivation

Support training with combined datasets

## Modification

1. add a new dataset wrapper `CombinedDataset`
2. add `KeypointConverter` to convert the channel order of `keypoints` and `keypoints_visible`
3. add unittests
4. add two example configs
  - `td-hm_hrnet-w32_8xb64-210e_coco-aic-256x192-merge.py` maps the keypoints of AIC to COCO format.\
  - `td-hm_hrnet-w32_8xb64-210e_coco-aic-256x192-combine.py` maps the keypoints of AIC & COCO to a combined format. 
5. modify default `dataset_meta` in `init_cfg` from test dataset to train dataset

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/models/heads/heatmap_heads/heatmap_head.py [hunk 1]
```diff
@@ -292,6 +292,8 @@ def predict(self,
         else:
             batch_heatmaps = self.forward(feats)
 
+        batch_heatmaps = self.select_output_indices(
```
Comment (Tau-J) [inline mmpose/models/heads/heatmap_heads/heatmap_head.py:295]: How about moving `select_output_indices` into `decode()` of `base_head.py`?

Comment (Ben-Louis) [reply mmpose/models/heads/heatmap_heads/heatmap_head.py:295]: `decode()` does not affect output heatmaps. There will be a mismatch between `pred_instance.keypoints` and `pred_fieds.heatmaps`

File: mmpose/models/pose_estimators/base.py [hunk 2]
```diff
@@ -17,13 +18,19 @@ class BasePoseEstimator(BaseModel, metaclass=ABCMeta):
             config of :class:`BaseDataPreprocessor`. Defaults to ``None``.
         init_cfg (dict | ConfigDict): The model initialization config.
             Defaults to ``None``
+        metainfo (dict): Meta information for dataset, such as keypoints
```
Comment (ly015) [inline mmpose/models/pose_estimators/base.py:21]: May need further descriptions, like the metainfo from the input data batch will be ignored if this argument is set. And it's better to provide a link to the document of the dataset metainfo here.

File: mmpose/models/pose_estimators/topdown.py [hunk 3]
```diff
@@ -178,6 +183,7 @@ def add_pred_to_datasample(self, batch_pred_instances: InstanceList,
         assert len(batch_pred_instances) == len(batch_data_samples)
         if batch_pred_fields is None:
             batch_pred_fields = []
+        kpt_indices = self.test_cfg.get('out_keypoint_indices', None)
```
Comment (ly015) [inline mmpose/models/pose_estimators/topdown.py:186]: I would suggest 'output_keypoint_indices' or 'pred_keypoint_indices'.

File: configs/body_2d_keypoint/topdown_heatmap/coco/hrnet_coco_aic.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+@article{wu2017ai,
+  title={Ai challenger: A large-scale dataset for going deeper in image understanding},
+  author={Wu, Jiahong and Zheng, He and Zhao, Bo and Li, Yixin and Yan, Baoming and Liang, Rui and Wang, Wenjia and Zhou, Shipei and Lin, Guosen and Fu, Yanwei and others},
+  journal={arXiv preprint arXiv:1711.06475},
+  year={2017}
+}
+```
+
+</details>
+
```
Comment (ly015) [inline configs/body_2d_keypoint/topdown_heatmap/coco/hrnet_coco_aic.md:49]: We should briefly explain `coco-aic-merge` and `coco-aic-combine`, like the former is to train a model that predicts both COCO and AIC keypoints, while the latter is to use AIC auxiliary data to train a COCO model.

And it's better also to add performance using a single dataset (COCO or AIC) into the table for comparison.

**Review by ly015 — APPROVED** [review]

**Pull request merged**